src/storage/cache/rpc/client.rs

```diff
@@ -0,0 +1,665 @@
+                        .store(current_timestamp, std::sync::atomic::Ordering::Release);
+                    // Update the received keep alive seq
+                    if let Ok(resp_type) = RespType::from_u8(header.op) {
+                        if let RespType::FileBlockResponse = resp_type {
+                            debug!("{:?} Received response header: {:?}", self, header);
+                            // Try to read to buffer
+                            match self.recv_len(header.len).await {
+                                Ok(()) => {}
+                                Err(err) => {
+                                    debug!(
+                                        "{:?} Failed to receive request header: {:?}",
+                                        self, err
+                                    );
+                                    break;
+                                }
+                            }
+
+                            // Take the packet task and recv the response
+                            let resp_buffer: &mut BytesMut = unsafe { &mut *self.resp_buf.get() };
+                            match self.packets_keeper.take_task(header_seq, resp_buffer).await {
+                                Ok(()) => {
+                                    debug!("{:?} Received response: {:?}", self, header_seq);
+                                }
+                                Err(err) => {
+                                    debug!("{:?} Failed to update task: {:?}", self, err);
+                                }
+                            }
+                        }
+                    } else {
+                        debug!("{:?} Invalid response type: {:?}", self, header.op);
+                        break;
+                    }
+                }
+                Err(err) => {
+                    debug!("{:?} Failed to receive request header: {:?}", self, err);
+                    break;
+                }
+            }
+        }
+    }
+
+    /// Get stream with mutable reference
+    #[allow(clippy::mut_from_ref)]
+    fn get_stream_mut(&self) -> &mut TcpStream {
+        // Current implementation is safe because the stream is only accessed by one thread
+        unsafe { &mut *self.stream.get() }
+    }
+}
+
+unsafe impl<P> Send for RpcClientConnectionInner<P> where P: Packet + Clone + Send + Sync + 'static {}
+
+unsafe impl<P> Sync for RpcClientConnectionInner<P> where P: Packet + Clone + Send + Sync + 'static {}
+
+/// The RPC client definition.
+#[derive(Debug)]
+pub struct RpcClient<P>
+where
+    P: Packet + Clone + Send + Sync + 'static,
+{
+    /// The timeout options for the client.
+    /// TODO:
+    #[allow(dead_code)]
+    timeout_options: ClientTimeoutOptions,
+    /// Inner connection for the client.
+    inner_connection: Arc<RpcClientConnectionInner<P>>,
+    /// Client ID for the connection
+    /// TODO:
+    #[allow(dead_code)]
+    client_id: AtomicU64,
+}
+
+impl<P> RpcClient<P>
```

Comment:
You should impl Drop for `RpcClient`, gracefully shutdown sender part, and notify recv_loop() to exit

Reply:
OK.